torch/csrc/Generator.cpp

```diff
@@ -1,272 +1,125 @@
-  0,                                           /* tp_itemsize */
-  THPGenerator_dealloc,                        /* tp_dealloc */
-  0,                                           /* tp_vectorcall_offset */
-  nullptr,                                     /* tp_getattr */
-  nullptr,                                     /* tp_setattr */
-  nullptr,                                     /* tp_reserved */
-  nullptr,                                     /* tp_repr */
-  nullptr,                                     /* tp_as_number */
-  nullptr,                                     /* tp_as_sequence */
-  nullptr,                                     /* tp_as_mapping */
-  nullptr,                                     /* tp_hash  */
-  nullptr,                                     /* tp_call */
-  nullptr,                                     /* tp_str */
-  nullptr,                                     /* tp_getattro */
-  nullptr,                                     /* tp_setattro */
-  nullptr,                                     /* tp_as_buffer */
-  Py_TPFLAGS_DEFAULT | Py_TPFLAGS_BASETYPE,    /* tp_flags */
-  nullptr,                                     /* tp_doc */
-  nullptr,                                     /* tp_traverse */
-  nullptr,                                     /* tp_clear */
-  nullptr,                                     /* tp_richcompare */
-  0,                                           /* tp_weaklistoffset */
-  nullptr,                                     /* tp_iter */
-  nullptr,                                     /* tp_iternext */
-  THPGenerator_methods,                        /* tp_methods */
-  THPGenerator_members,                        /* tp_members */
-  THPGenerator_properties,                     /* tp_getset */
-  nullptr,                                     /* tp_base */
-  nullptr,                                     /* tp_dict */
-  nullptr,                                     /* tp_descr_get */
-  nullptr,                                     /* tp_descr_set */
-  0,                                           /* tp_dictoffset */
-  nullptr,                                     /* tp_init */
-  nullptr,                                     /* tp_alloc */
-  THPGenerator_pynew,                          /* tp_new */
-};
-
-bool THPGenerator_init(PyObject *module)
-{
-  THPGeneratorClass = (PyObject*)&THPGeneratorType;
-  if (PyType_Ready(&THPGeneratorType) < 0)
-    return false;
-  Py_INCREF(&THPGeneratorType);
-  PyModule_AddObject(module, "Generator", (PyObject *)&THPGeneratorType);
-  return true;
-}
-
-void set_pyobj(const Generator& self, PyObject* pyobj) {
-  TORCH_CHECK(self.defined(), "cannot call set_pyobj() on undefined generator");
-  self.set_pyobj(pyobj);
+} // namespace
+
+void initGeneratorBindings(PyObject* module) {
+  py::options options;
+  options.disable_user_defined_docstrings();
+  options.disable_function_signatures();
+
+  py::class_<Generator>(module, "Generator", py::is_final())
```

Comment:
Yeah `py::is_final` is [a fairly recent addition](https://github.com/pybind/pybind11/releases/tag/v2.6.0). I think we can at least skip it here since in the current version of PyTorch it's possible to subclass `torch.Generator` in Python, but I'll need to come up with a solution for other binding classes which disallow it (e.g. `torch.device`, `torch.dtype`).

Reply:
Done.